internal/ostree/ostree.go

```diff
@@ -65,47 +72,56 @@ func ResolveRef(location, ref string) (string, error) {
 	return parent, nil
 }
 
-// ResolveParams resolves all necessary missing parameters in the given struct:
-// it sets the defaultRef if none is provided and resolves the parent commit if
-// a URL and Ref are provided. If there is an error, it will be of type
-// InvalidParameterError or ResolveRefError (from the ResolveRef function)
-func ResolveParams(params RequestParams, defaultRef string) (RequestParams, error) {
-	resolved := RequestParams{}
-	resolved.Ref = params.Ref
-	// if ref is not provided, use distro default
-	if resolved.Ref == "" {
-		resolved.Ref = defaultRef
-	} else if !VerifyRef(params.Ref) { // only verify if specified in params
-		return resolved, NewRefError("Invalid ostree ref %q", params.Ref)
+// ResolveParams resolves the ostree request parameters into the necessary ref
+// for the image build pipeline and a commit (checksum) to be fetched.
+//
+// If a URL is defined in the RequestParams, the checksum of the Parent ref is
+// resolved, otherwise the checksum is an empty string. Specifying Parent
+// without URL results in a ParameterComboError. Failure to resolve the
+// checksum results in a ResolveRefError.
+//
+// If Ref is not specified in the RequestParams, the defaultRef is used.
+// If Parent is not specified in the RequestParams, the value of Ref is used
+// (which will be defaultRef if neither is specified).
+//
+// If any ref (Ref or Parent) is malformed, the function returns with a RefError.
+func ResolveParams(params RequestParams, defaultRef string) (ref string, checksum string, err error) {
```

Comment:
Nitpick: you can simplify the return values declaration to `(ref, checksum string, err error)`